src/reason_pprint_ast.ml

```diff
@@ -2376,29 +2395,26 @@ class printer  ()= object(self:'self)
 
   method type_with_label (label, ({ptyp_desc} as c)) =
     match label with
-      | "" ->  self#non_arrowed_non_simple_core_type c (* otherwise parenthesize *)
-      | s  ->
-          if s.[0]='?' then
-            let len = String.length s - 1 in
-            let lbl = String.sub s 1 len in
-            match ptyp_desc with
-              | Ptyp_constr ({txt}, l) ->
-                  assert (is_predef_option txt);
-                  let everythingButQuestion =
-                    formatLabeledArgument
-                      (atom lbl)
-                      ""
-                      (makeList
-                         ~postSpace:true
-                         ~break:IfNeed
-                         ~inline:(true, true)
-                         (* Why not support aliasing here? *)
-                         (* I don't think you'll have more than one l here. *)
-                         (List.map (self#non_arrowed_non_simple_core_type) l)
-                      ) in
-                    makeList [everythingButQuestion; atom "?"]
-              | _ -> failwith "invalid input in print_type_with_label"
-          else formatLabeledArgument (atom s) "" (self#non_arrowed_non_simple_core_type c)
+    | Nolabel ->  self#non_arrowed_non_simple_core_type c (* otherwise parenthesize *)
+    | Labelled s -> formatLabeledArgument (atom s) "" (self#non_arrowed_non_simple_core_type c)
+    | Optional lbl ->
+      match ptyp_desc with
+        | Ptyp_constr ({txt}, l) ->
+            assert (is_predef_option txt);
+            let everythingButQuestion =
+              formatLabeledArgument
+                (atom lbl)
+                ""
+                (makeList
+                   ~postSpace:true
+                   ~break:IfNeed
+                   ~inline:(true, true)
+                   (* Why not support aliasing here? *)
```

Comment:
Feel free to delete this comment. I don't know what it means anymore, but it is likely not applicable any more.